core/src/main/java/org/apache/calcite/rel/rules/materialize/MaterializedViewOnlyAggregateRule.java

```diff
@@ -69,18 +72,25 @@ public MaterializedViewOnlyAggregateRule(RelOptRuleOperand operand,
   }
 
   /** Rule configuration. */
-  public interface Config extends MaterializedViewAggregateRule.Config {
-    Config DEFAULT = create(RelFactories.LOGICAL_BUILDER);
+  @Value.Immutable
+  public interface Config extends MaterializedViewAggregateRule.Config<Config> {
+    Config DEFAULT = ImmutableMaterializedViewOnlyAggregateRule.Config.builder()
+        .withRelBuilderFactory(RelFactories.LOGICAL_BUILDER)
+        .withGenerateUnionRewriting(true)
+        .withUnionRewritingPullProgram(null)
+        .withFastBailOut(false)
+        .build();
 
     static Config create(RelBuilderFactory relBuilderFactory) {
-      return MaterializedViewAggregateRule.Config.create(relBuilderFactory)
+
+      return ImmutableMaterializedViewOnlyAggregateRule.Config.builder()
```

Comment:
You can also do build from `DEFAULT` instance:

```java
ImmutableMaterializedViewOnlyAggregateRule.Config.builder()
.from(DEFAULT)
.withRelBuilderFactory(relBuilderFactory)
.withOperandSupplier(b -> b.operand(Aggregate.class).anyInputs())
.withDescription("MaterializedViewAggregateRule(Aggregate)")
.build();
```

Reply:
Thanks. In most cases I use builders to build the default so from(DEFAULT) doesn't really work.